lib/managementClient.ts

```diff
@@ -0,0 +1,231 @@
+   * @param entityPath - The name/path of the entity (hub name) for which the management request needs to be made.
+   */
+  constructor(public entityPath: string) {
+    this.entityPath = entityPath;
+  }
+
+  /**
+   * Provides the eventhub runtime information.
+   * @method getHubRuntimeInformation
+   * @param {Connection} connection - The established amqp connection
+   * @returns {Promise<EventHubRuntimeInformation>}
+   */
+  async getHubRuntimeInformation(connection: any): Promise<EventHubRuntimeInformation> {
+    const info: any = await this._makeManagementRequest(connection, Constants.eventHub);
+    const runtimeInfo: EventHubRuntimeInformation = {
+      path: info.name,
+      createdAt: new Date(info.created_at),
+      partitionCount: info.partition_count,
+      partitionIds: info.partition_ids,
+      type: info.type
+    };
+    debug(`[%s] The hub runtime info is.`, connection.options.id, runtimeInfo);
+    return runtimeInfo;
+  }
+
+  /**
+   * Provides an array of partitionIds.
+   * @method getPartitionIds
+   * @param {Connection} connection - The established amqp connection
+   * @returns {Promise<Array<string>>}
+   */
+  async getPartitionIds(connection: any): Promise<Array<string>> {
+    let runtimeInfo = await this.getHubRuntimeInformation(connection);
+    return runtimeInfo.partitionIds;
+  }
+
+  /**
+   * Provides information about the specified partition.
+   * @method getPartitionInformation
+   * @param {Connection} connection - The established amqp connection
+   * @param {(string|number)} partitionId Partition ID for which partition information is required.
+   */
+  async getPartitionInformation(connection: any, partitionId: string | number): Promise<EventHubPartitionRuntimeInformation> {
+    if (!partitionId || (partitionId && typeof partitionId !== "string" && typeof partitionId !== "number")) {
+      throw new Error("'partitionId' is a required parameter and must be of type: 'string' | 'number'.");
+    }
+    const info: any = await this._makeManagementRequest(connection, Constants.partition, partitionId);
+    const partitionInfo: EventHubPartitionRuntimeInformation = {
+      beginningSequenceNumber: info.begin_sequence_number,
+      hubPath: info.name,
+      lastEnqueuedOffset: info.last_enqueued_offset,
+      lastEnqueuedTimeUtc: new Date(info.last_enqueued_time_utc),
+      lastSequenceNumber: info.last_enqueued_sequence_number,
+      partitionId: info.partition,
+      type: info.type
+    };
+    debug(`[%s] The partition info is: ${partitionInfo}.`, connection.options.id);
+    return partitionInfo;
+  }
+
+  /**
+   * Closes the AMQP management session to the Event Hub for this client,
+   * returning a promise that will be resolved when disconnection is completed.
+   * @return {Promise<void>}
+   */
+  async close(): Promise<void> {
+    try {
+      if (this._mgmgtReqResLink) {
+        await rheaPromise.closeSession(this._mgmgtReqResLink.session);
```

Comment:
fwiw this seems like it falls under the same question of whether closing the session fully cleans up the underlying links